### PR TITLE
[doc][gradle plugin] publishing to maven

### PR DESCRIPTION
**Pull request by vvlevchenko**
depends on #1688

### CODE REVIEW

File: MAVEN.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#Publishing to maven.
```
Comment (StefMa) [inline MAVEN.md]: Put an space between `#` and `P`.
Otherwise Github don't render it correctly:
bildschirmfoto 2018-06-14 um 16 29 38

File: MAVEN.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+Publishing the kotlin native artifacts borrows mechanisms introduced in Gradle Native support, e.g. gradle metadata feature,
```
Comment (StefMa) [inline MAVEN.md]: Should you say `Kotlin/Native`? 🤔

Comment (StefMa) [reply MAVEN.md]: I would reword it to something like:
> Publishing Kotlin/Native artifacts depends on a mechanisms which was introduced with the Gradle Native support, e.g. the Gradle metadata feature [....]

File: MAVEN.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+thus there are some additional steps are required. First of all gradle version it shouldn't be less then gradle version of
```
Comment (StefMa) [inline MAVEN.md]: Gradle - in uppercase..
(Everywhere where you mention Gradle 😉)

File: MAVEN.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+#Publishing to maven.
+
+Publishing the kotlin native artifacts borrows mechanisms introduced in Gradle Native support, e.g. gradle metadata feature,
+thus there are some additional steps are required. First of all gradle version it shouldn't be less then gradle version of 
+kotlin native plugin dependens on (currently gradle-4.7). before gradle-5.0 feature GRADLE_METADATA should be enabled for build.
```
Comment (StefMa) [inline MAVEN.md]: Without the minus in the Gradle version: `Gradle 4.7` (same for Gradle 5.0)

Comment (StefMa) [inline MAVEN.md]: I think it make sense to link a URL to some kind of Gradle docu where the feature "GRADLE_METADATA" is - well - documented...?!

File: MAVEN.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+e.g. in settings.gradle
+````
+enableFeaturePreview('GRADLE_METADATA')
+````
+
+Some maven repositories requires some declarations in `pom` files, that should be presents in all auxilary `pom` files (
+platform x build types). To meet this requirement kotlin native plugin offers special syntax:
```
Comment (StefMa) [inline MAVEN.md]: > To meet this requirement the Kotlin/Native Plugin have the following syntax to do it:

File: MAVEN.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+                 def root = asNode()
+                 root.appendNode('name', 'libcurl interop library')
+                 root.appendNode('description', 'A library providing interoperability with host libcurl')
+             }
+         }
+     }
+ }
+
+ ````
+ In this example `name` and `description` tags will be added to each `pom` file generated for _libcurl_ published artifact.
```
Comment (StefMa) [inline MAVEN.md]: > In this example `name` and `description` tags will be added to each generated `pom` file for each _libcurl_ published artifact.

File: GRADLE_PLUGIN.md [hunk 7]
```diff
@@ -599,3 +599,38 @@ tables below.
         // Use the ':foo' project as a common project for multiplatform build.
         expectedBy project(':common')
     }
+
+## Publishing to maven.
```
Comment (olonho) [inline GRADLE_PLUGIN.md]: maven -> Maven
